### DISCUSSION TITLE
External Component to control lock icon

### DISCUSSION THREAD

**Discussion by viveknandhan527** [react-querybuilder/react-querybuilder#467] [Q&A] [answered]
Hi Team 

I would like to add EDIT button outside query builder and based on user action query builder need to be editable. Appreciate for help and for the steps to add Edit button (Lock) out side the query builder

**Comment by jakeboone02** [comment, marked as answer]
[This codesandbox](https://codesandbox.io/p/sandbox/u127lx?file=%2Fsrc%2FApp.tsx), based on the [basic-ts example](https://github.com/react-querybuilder/react-querybuilder/tree/main/examples/basic-ts), has a toggle button at the top to switch the [`disabled` prop](https://react-querybuilder.js.org/docs/components/querybuilder#disabled) between `true` and `false`. When `true`, all controls in the query builder are disabled.

![image](https://user-images.githubusercontent.com/366438/220647227-85bd40a6-7abf-45d6-8ae6-7c0526bd0e06.png)

Hope that points you in the right direction.

**Reply by viveknandhangelella** [reply]
thank you @jakeboone02  it worked
